### ISSUE TITLE
Discussion: Enabling motion tracking for inline sessions.

### ISSUE THREAD

**Issue by toji** [immersive-web/editor-collab#15]
We need to have a discussion about exactly how we want to have developers indicate the need for motion tracking on an inline session. I think the assumption has been that the creation of a stationary reference space would be the signal that the developer wants to engage in any available tracking, but I'm concerned about how that interacts with the rest of our user consent flow. Specifically, if enabling tracking requires any sort of user consent and the creation of the reference space is what triggers it you may inadvertently end up in a situation where you're presenting multiple dialogs to the user (some during session creation and some during reference space creation). It may also imply that the space creation be gated on user activation so that we're not showing users consent dialogs at random. (That could be worked around by having the consent request be shown inline on the canvas or something, but that's a UA descision.) Additionally, it would be a break in behavior between immersive sessions, which assume that you can create reference spaces without user input of any sort, and requires a divergent code path for inline which we generally want to reduce.

It seems to me like a reasonable alternative would be to say that inline sessions get the same guarantees about stationary reference space availability that other session types get, but those reference spaces are inert unless you've requested an "inline-tracking" feature (or similar) at session creation time. That would appear to mesh with the immersive use cases better and you get an added benefit of not needing to do additional transform math on the viewer poses just to get them to line up with a floor-level scene just because you didn't actually need tracking.

I'm sure I'm overlooking something in my overly-simplistic assessment here, which is why I'd like to discuss it prior to making any actual proposals.

**Comment by NellWaliczek** [comment]
Yeah, I very strongly agree that the creation of an `XRReferenceSpace` should not ever pop UI.  

And I'd also been assuming that we'd address this issue via a feature request, and is a big part of why I'm not super comfortable with treating `XRReferenceSpace`s as different in the desired/required pattern.  For example, I'd like to see developers decide for themselves if motion tracking is a desired features (they'll support mouse/keyboard/touch method for changing the viewer's location) or required.

In other words, I don't think the feature that should be requested is `inline-tracking` but instead it should be `stationary-reference-space`.

**Comment by cwilso** [comment]
Hey, shouldn't this issue be over in the core spec issues?

**Comment by toji** [comment]
I also felt there was a decent chance that I had just misunderstood an element of the spatial tracking work that had already been written, so I wanted to sanity check with Nell prior to making noise on the core issue list. I'm perfectly happy to create issues on the webxr repo once I've established what's actually missing. :D

**Comment by toji** [comment]
Concrete next step that came out of today's editors sync: We want to create a new reference space type (not subtype) that is "poseless", in that it only returns identity poses. This type should only be allowed in inline sessions, and is primarily useful when paired with the `originOffset` attribute recently added to the `XRReferenceSpace` to offer click/drag/touch interactions. The `stationary` subtypes can still be created for inline sessions, but can require user consent. (Same for an expected future `diorama` type).
